client/src/shared/state/Inventory.js

```diff
@@ -104,16 +107,155 @@ class Inventory {
+                if (updateValue >= 0) {
+                    this.keyToSlotIndex[key] = updateValue;
+                    if (this.saveSession === true) {
+                        this.saveHotbar();
+                    }
+                }
+            }
+        });
+    }
+
+    updateMapHotBarKeys(evictKey) {
+        Object.keys(this.keyToSlotIndex).forEach((key, index) => {
+            if (key >= evictKey) {
+                const updateKey = key - 1;
+                if (updateKey >= 0) {
+                    this.keyToSlotIndex[updateKey] = this.keyToSlotIndex[key];
+                    delete this.keyToSlotIndex[key];
+                }
+            }
+        });
+    }
+
+    updateMapSlotIndex(evictValue) {
+        Object.keys(this.keyToSlotIndex).forEach((key, index) => {
+            if (this.keyToSlotIndex[key] >= evictValue) {
+                const newValue = this.keyToSlotIndex[key] - 1;
+                if (newValue >= 0) {
+                    this.keyToSlotIndex[key] = newValue;
+                }
+            }
+        });
+    }
+
+    removeFromHotbar(itemConfig, removedInventory = false) {
+        const getKeys = Object.keys(this.keyToSlotIndex);
+        const evictKey = getKeys.find((key) => this.keyToSlotIndex[key] === itemConfig.slotIndex);
+        const evictValue = this.keyToSlotIndex[evictKey];
+        if (typeof evictKey === "undefined") {
+            this.updateSlotIndex(itemConfig.slotIndex);
+            return;
+        }
+        delete this.keyToSlotIndex[evictKey];
+        this.updateMapHotBarKeys(evictKey);
+        if (removedInventory) {
+            this.updateMapSlotIndex(evictValue);
+        }
         this.hotbar = this.hotbar.filter((eachItem) => eachItem !== itemConfig);
 
         PubSub.publish(HOTBAR_ITEM);
         PubSub.publish(MODIFY_INVENTORY_ITEM);
+        if (this.saveSession === true) {
+            this.saveHotbar();
+        }
+    }
+
+    saveHotbar() {
+        const sessionIDMap = "saveHotBar";
+        const localStorageID = this.userName + sessionIDMap;
+        window.localStorage.removeItem(localStorageID);
+        window.localStorage.setItem(localStorageID, JSON.stringify(this.keyToSlotIndex));
+        // Used to debug
+        // Utils.message(this.keyToSlotIndex);
+    }
+
+    defaultHotBar() {
+        this.items.forEach((itemConfig) => {
+            this.addToHotbar(itemConfig);
+        });
+    }
+
+    initializeHotbar() {
+        const saveDataSize = Object.keys(this.keyToSlotIndex).length;
+        this.items.forEach((itemConfig) => {
+            if (this.hotbar.length >= saveDataSize) {
+                return;
+            }
+            this.hotbar.push(itemConfig);
+        });
+    }
+
+    loadHotbar(playerID, isLoggedIn) {
+        if (isLoggedIn === true) {
+            this.saveSession = true;
+        }
+        this.userName = playerID;
+        const sessionIDMap = "saveHotBar";
+        const localStorageID = this.userName + sessionIDMap;
```

Comment:
if you move `sessionIDMap` into a class field, then `localStorageID` could be changed to a class method also (moved to the class definition at top of file)
```suggestion
        function localStorageID() {return this.userName + this.sessionIDMap};
```